doc/FuzzTestsProgrammerManual.md

```diff
@@ -0,0 +1,47 @@
+# Fuzz Testing on xively-client-c IoT library
```

Comment:
two things regarding the file itself:
1 - naming, isn't this the first CamelCase named file in the repo? What about fuzz_tests_programmer_manual.md, isn't his more libxively style?
2 - location. I don't think this programming manual is for the same interest level visitors, so I wouldn't put this on the same level as porting guide and user guide. I'd put this under doc/dev ?
And I've found another camel cased filename doc/res